### PR TITLE
No bg window

### PR DESCRIPTION
**Pull request by danlipsa**
This are changes for no window pop up for vcs-js. They also make behaviour for
https://github.com/UV-CDAT/vcs/issues/228
better.

### CODE REVIEW

**Comment by mattben** [comment]
Thank you @danlipsa

**Comment by doutriaux1** [comment]
@danlipsa looking at circleci errors, I think you didn't start from master. It should find a vtk-cdat I think I removed the constraint on vtk-cdat in master.  see: https://github.com/UV-CDAT/vcs/blob/master/ci-support/circleci_mac_dep.sh#L5

File: vcs/VTKPlots.py [hunk 1]
```diff
@@ -423,7 +418,13 @@ def clear(self, render=True):
     def createDefaultInteractor(self, ren=None):
         defaultInteractor = self.renWin.GetInteractor()
         if defaultInteractor is None:
-            defaultInteractor = vtk.vtkRenderWindowInteractor()
+            if self.bg:
```
Comment (aashish24) [inline vcs/VTKPlots.py]: looks good @danlipsa 

quick question what happens if we change the bg=xx multiple times..does this function gets called every time

Comment (danlipsa) [reply vcs/VTKPlots.py]: The function is called only if the render window is not None so it won't be called every time.

**Comment by danlipsa** [comment]
@doutriaux1 Any ideas on the travis failure? test_vcs_no_slowdown passes on my machine with both hardware driver and mesa. Also, this commit should not affect this test.

**Comment by doutriaux1** [comment]
@danlipsa sometimes we just need to restart the travis build. I'll try it now.

**Comment by danlipsa** [comment]
@doutriaux1 Nice, ready to merge?

File: vcs/VTKPlots.py [hunk 2]
```diff
@@ -37,7 +37,6 @@ def __init__(self, parent):
         self.AddObserver(
             "LeftButtonReleaseEvent",
             parent.leftButtonReleaseEvent)
-        self.AddObserver("ModifiedEvent", parent.configureEvent)
```
Comment (doutriaux1) [inline vcs/VTKPlots.py]: @danlipsa did you try it on MAc as well (I'm about to anyway). I remember WAY back then that there were some issues with Mac not being able to differentiate between ModifiedEvent and ConfigureEvent or something like that.

File: vcs/VTKPlots.py [hunk 3]
```diff
@@ -278,10 +276,6 @@ def leftButtonReleaseEvent(self, obj, event):
     def configureEvent(self, obj, ev):
         if not self.renWin:
             return
-        cursor = self.renWin.GetCurrentCursor()
-        if sys.platform == "darwin" and ev == "ModifiedEvent" and cursor != self.oldCursor:
```
Comment (doutriaux1) [inline vcs/VTKPlots.py]: that line is probably here because of the reason I mentioned above.

File: vcs/VTKPlots.py (hunk 1 above)
Comment (doutriaux1) [reply vcs/VTKPlots.py]: @aashish24 good p[oint. @danlipsa should we (in a spearate PR) lock self.bg once something has been plotted?

File: vcs/VTKPlots.py (hunk 2 above)
Comment (danlipsa) [reply vcs/VTKPlots.py]: Yes, I did tried it on mac, in vcs-js context.

Comment (danlipsa) [reply vcs/VTKPlots.py]: It works nicely. Using plane vcs, I get a window without a frame that I cannot resize. Is this a rencent occurence?

**Pull request merged**